sqlx-core/src/testing/mod.rs

```diff
@@ -28,19 +30,22 @@ pub trait TestSupport: Database {
 
     fn cleanup_test(db_name: &str) -> BoxFuture<'_, Result<(), Error>>;
 
-    /// Cleanup any test databases that are no longer in-use.
-    ///
-    /// Returns a count of the databases deleted, if possible.
-    ///
-    /// The implementation may require `DATABASE_URL` to be set in order to manage databases.
-    /// The user credentials it contains must have the privilege to create and drop databases.
-    fn cleanup_test_dbs() -> BoxFuture<'static, Result<Option<usize>, Error>>;
-
     /// Take a snapshot of the current state of the database (data only).
     ///
     /// This snapshot can then be used to generate test fixtures.
     fn snapshot(conn: &mut Self::Connection)
         -> BoxFuture<'_, Result<FixtureSnapshot<Self>, Error>>;
+
+    /// Generate a unique database name for the given test path.
+    fn db_name(args: &TestArgs) -> String {
+        let mut hasher = Sha512::new();
+        hasher.update(args.test_path.as_bytes());
+        let hash = hasher.finalize();
+        let hash = STANDARD.encode(&hash[..39]);
```

Comment:
The `STANDARD` alphabet of Base64 is problematic if unquoted because it uses `+` and `/` as the two non-alphanumeric characters.

I think for this application I'd like to use the `URL_SAFE` alphabet instead: https://docs.rs/base64/latest/base64/alphabet/constant.URL_SAFE.html

It will still require quoting of the database name due to the hyphen but it means it can be used in a database URL without percent-encoding.